pkg/providers/vsphere/defaults.go

```diff
@@ -99,26 +101,69 @@ func (d *Defaulter) setupDefaultTemplate(ctx context.Context, spec *Spec, machin
 	if err != nil {
-		return fmt.Errorf("getting template details: %v", err)
+		return fmt.Errorf("getting template snapshot details: %v", err)
 	}
 
-	if !templateHasSnapshot {
-		logger.Info("Warning: Your VM template has no snapshots. Defaulting to FullClone mode. VM provisioning might take longer.")
-		if machineConfig.Spec.DiskGiB < 20 {
-			logger.Info("Warning: VSphereMachineConfig DiskGiB cannot be less than 20. Defaulting to 20.")
-			machineConfig.Spec.DiskGiB = 20
+	// if the user didn't set the cloneMode, determine the clone mode based on the following rules:
+	// - if the template has snapshots and the template diskSize matches the machineConfig DiskGiB, default to linked clone
+	// - else use full clone
+	// if the user set the cloneMode to fullClone, use full clone
+	// if the user set the cloneMode to linkedClone, use linked clone if the following properties are met:
+	// - if the template has snapshots
+	// - if the template's disk size matches the VSphereMachineConfig diskGiB
+
+	switch machineConfig.Spec.CloneMode {
+	case "":
+		if templateHasSnapshot && machineConfig.Spec.DiskGiB == diskSize {
+			logger.V(3).Info("CloneMode not set, defaulting to linkedClone", "VSphereMachineConfig", machineConfig.Name)
+			machineConfig.Spec.CloneMode = anywherev1.LinkedClone
+		} else {
+			logger.V(3).Info("CloneMode not set, defaulting to fullClone", "VSphereMachineConfig", machineConfig.Name)
+			machineConfig.Spec.CloneMode = anywherev1.FullClone
+		}
+
+	case anywherev1.FullClone:
+		// do nothing
+
+	case anywherev1.LinkedClone:
+		if !templateHasSnapshot {
+			return fmt.Errorf(
+				"linkedClone can not be used for VSphereMachineConfig %s has the template %s has no snapshots. Ensure that the template has snapshots or change the cloneMode to fullClone",
+				machineConfig.Name,
+				machineConfig.Spec.Template,
+			)
 		}
-	} else if machineConfig.Spec.DiskGiB != 25 {
-		logger.Info("Warning: Your VM template includes snapshot(s). LinkedClone mode will be used. DiskGiB cannot be customizable as disks cannot be expanded when using LinkedClone mode. Using default of 25 for DiskGiBs.")
-		machineConfig.Spec.DiskGiB = 25
+		if machineConfig.Spec.DiskGiB != diskSize {
+			return fmt.Errorf(
+				"diskGiB can not be customized for VSphereMachineConfig %s when using linkedClone. Either change the cloneMode to fullClone or change the diskGiB to match %s's disk size of %d GiB",
+				machineConfig.Name,
+				machineConfig.Spec.Template,
+				diskSize,
+			)
+		}
+
+	default:
+		return fmt.Errorf("cloneMode %s is not supported for VSphereMachineConfig %s. Supported clone modes: [%s, %s]", machineConfig.Spec.CloneMode, machineConfig.Name, anywherev1.LinkedClone, anywherev1.FullClone)
 	}
 
 	return nil
 }
 
+func (d *Defaulter) setDiskSizeDefaults(machineConfig *anywherev1.VSphereMachineConfig) {
+	if machineConfig.Spec.DiskGiB < 20 {
+		logger.Info("Warning: VSphereMachineConfig DiskGiB cannot be less than 20. Defaulting to 20.")
```

Comment:
Is there not a warning method/function we can use? Prepending `Warning` typically leads to inconsistencies as people do `warn:` `WARN:` etc over time.

Reply:
I don't think we have this yet. I just moved this call from here https://github.com/aws/eks-anywhere/blob/main/pkg/providers/vsphere/defaults.go#L110-L113
I can create an issue to track this separately since it will require updating the other log messages not included as part of this PR